Orders index: Closes #19

dphilla opened the pull request:
Completes user can see all of their previous orders. 
Added comments for where things will probably be changed in story 13.

MikelSage commented on app/views/orders/index.html.erb:
```diff
@@ -0,0 +1,5 @@
+<h1> Here are your past orders: </h1>
+
+<h1>  <% @orders.each do |order| %> </h1>
+<p> Your order from <%= order.resolved_at.to_s %> </p>  <!--- These could become links in story 13 --->
```
Definitely like that idea. Nice use of comments here to let us know your thoughts!

jimszalew replied:
Thanks for making it easy for me when I pick up #13!

MikelSage commented on config/routes.rb:
```diff
@@ -10,6 +10,9 @@
 
+  resources :users, only: :show do
+    resources :orders, only: :index
+  end
```
![image](https://user-images.githubusercontent.com/26507085/28741343-58566dba-73d1-11e7-80d5-a0e935546f50.png)
Nice nest!

MikelSage replied:
Looking back at the user story for this one it says the path should be slash orders, but IIRC this will actually make the path '/users/:id/orders'.

MikelSage commented on spec/features/orders/user_orders_index_spec.rb:
```diff
@@ -0,0 +1,16 @@
+    user = create(:user)
+    user_order1 = create(:order, user_id: user.id)
+    user_order2 = create(:order, user_id: user.id)
+
```
I think we might be able to make this a bit more robust by stubbing out the current_user, and then using the current_user method in the order's controller action instead of finding by the id passed in.

dphilla replied:
yeah, that's a really good point. I need to get some paths in there to test for an unauthenticated user trying to look at someone else's orders.

dphilla commented:
I added the current_user check to the orders controller and changed the routes to reflect the requirements in the spec, but I'm really not sure if it could be done in a better way. Thoughts?

MikelSage commented on app/controllers/orders_controller.rb:
```diff
@@ -1,8 +1,14 @@
-    @user = User.find(params[:user_id])
-    @orders = @user.orders
+
+    if @user = current_user
```
Yeah, this is pretty much what I was thinking of, the only way it might be cleaned up is 

```
if current_user
  @orders = current_user.orders
else
  #flash message and redirect
end
```

Otherwise I think it's ready to merge

MikelSage commented on app/controllers/orders_controller.rb:
```diff
@@ -0,0 +1,14 @@
+
+  def index
+
+    if @user = current_user
```
Pretty much what I was thinking of, only thing might be to refactor it a bit to: 

```
if current_user
  @orders = current_user.orders
else
  #flash message and redirect
end
```
Otherwise I think it's ready to merge.
